server/graphql/v2/mutation/PaymentMethodMutations.js

```diff
@@ -1,39 +1,53 @@
-import { GraphQLBoolean, GraphQLNonNull, GraphQLString } from 'graphql';
+import { GraphQLBoolean, GraphQLNonNull, GraphQLObjectType, GraphQLString } from 'graphql';
 import { pick } from 'lodash';
 
 import models from '../../../models';
 import { setupCreditCard } from '../../../paymentProviders/stripe/creditcard';
-import { Forbidden, ValidationFailed } from '../../errors';
+import { Forbidden } from '../../errors';
 import { AccountReferenceInput, fetchAccountWithReference } from '../input/AccountReferenceInput';
 import { CreditCardCreateInput } from '../input/CreditCardCreateInput';
-import { PaymentMethodCreateInput } from '../input/PaymentMethodCreateInput';
+import { fetchPaymentMethodWithReference, PaymentMethodReferenceInput } from '../input/PaymentMethodReferenceInput';
 import { PaymentMethod } from '../object/PaymentMethod';
+import { StripeError } from '../object/StripeError';
+
+const CreditCardWithStripeError = new GraphQLObjectType({
+  name: 'CreditCardWithStripeError',
+  fields: {
+    paymentMethod: {
+      type: new GraphQLNonNull(PaymentMethod),
+      description: 'The payment method created',
+    },
+    stripeError: {
+      type: StripeError,
+      description: 'This field will be set if there was an error with Stripe during strong customer authentication',
+    },
+    shouldBeSaved: {
+      type: GraphQLBoolean,
+      description: 'If the Payment Method needs to be marked as saved again during confirmCreditCard',
+    },
+  },
+});
 
 const addCreditCard = {
-  type: new GraphQLNonNull(PaymentMethod),
+  type: GraphQLNonNull(CreditCardWithStripeError),
```

Comment:
Just flagging that this is a breaking change.
- Either we deploy frontend & API at the same time (but there's still a risk)
- Or we preserve the legacy behavior & return type in `addStripeCreditCard`, so people with an outdated frontend will not see any crash

Reply:
You're right, sorry! Got a little too into the code cleaning 😳 I will add the deprecation back.

Reply:
Changing as discussed on Slack:

* adding back `addStripeCreditCard` as the mutation used to be, just so nothing breaks coming from the frontend when we deploy the API PR
* keeping these changes to `addCreditCard`
* then will write a follow up PR to remove `addStripeCreditCard` again

Both mutations are only used in the recurring contributions page now so this will be fine.